src/main/java/mekanism/common/tile/TileEntityTeleporter.java

```diff
@@ -162,15 +165,44 @@ private void cleanTeleportCache() {
     }
 
     private void resetBounds() {
-        teleportBounds = new AxisAlignedBB(getPos(), getPos().add(1, 3, 1));
+        switch(frameDirection) {
+        case UP:
+            teleportBounds = new AxisAlignedBB(getPos(), getPos().add(1, 3, 1));
+            break;
+        case DOWN:
+            teleportBounds = new AxisAlignedBB(getPos(), getPos().add(1, -3, 1));
+            break;
+        case EAST:
+            teleportBounds = new AxisAlignedBB(getPos(), getPos().add(3, 1, 1));
+            break;
+        case WEST:
+            teleportBounds = new AxisAlignedBB(getPos(), getPos().add(-3, 1, 1));
+            break;
+        case NORTH:
+            teleportBounds = new AxisAlignedBB(getPos(), getPos().add(1, 1, -3));
+            break;
+        case SOUTH:
+            teleportBounds = new AxisAlignedBB(getPos(), getPos().add(1, 1, 3));
+            break;
+        default:
+            teleportBounds = null;
+            break;
```

Comment:
This should all be indented once more rather than being in line with the switch statement itself

Reply:
this is how eclipse formats it, but i can change that.

Reply:
Eclipse's default formatting is pretty bad